sdk/containerregistry/azure-containerregistry/azure/containerregistry/_models.py

```diff
@@ -105,6 +108,16 @@ def __init__(self, **kwargs):
         if self.writeable_properties:
             self.writeable_properties = ContentProperties._from_generated(self.writeable_properties)
 
+        for data in ArtifactArchitecture:
+            if data.value == self.architecture:
+                self.architecture = data
+                break
```

Comment:
You should be able to do `ArtifactArchitecture(self.architecture)` if you want to convert the string to the enum type. Is this an input or result model? I feel like python users prefer to work with strings but ignore me if this design decision was already made.

Reply:
This is a result model. The `ArtifactArchitecture` inherits from str and Enum, users should be able to work with strings with this model.